pyquil/api/_qpu.py

```diff
@@ -168,10 +168,17 @@ def run(self, run_priority: Optional[int] = None):
         else:
             bitstrings = None
 
-        self._bitstrings = bitstrings
+        self._memory_results = dict()
+        for aref, vals in self._variables_shim.items():
+            self._memory_results[aref] = [vals] * ro_sources[0].shape[0]
+        self._memory_results["ro"] = bitstrings
         self._last_results = results
+
         return self
 
+    def read_memory(self, *, region_name: str):
+        return super().read_memory(region_name=region_name)
```

Comment:
Why add this method if all it does is call the superclass method of the same name?

Reply:
bitten by incremental development